### PR TITLE
Feature/notification settings 2.0

### PR DESCRIPTION
**Pull request by STLVRTX**

### CODE REVIEW

File: examples/e2e/module/user_email_contact.tf [hunk 1]
```diff
@@ -0,0 +1,6 @@
+resource "ilert_user_email_contact" "this" {
+  target = "${var.name}@fake.com"
```
Comment (yacut) [inline examples/e2e/module/user_email_contact.tf:2]: we always use `@example.com`

File: examples/e2e/module/user_phone_number_contact.tf [hunk 2]
```diff
@@ -0,0 +1,7 @@
+resource "ilert_user_phone_number_contact" "this" {
+  region_code = "DE"
+  target      = "+4915123456789"
```
Comment (yacut) [inline examples/e2e/module/user_phone_number_contact.tf:3]: where is this phone number coming from?

File: examples/escalation_policy/main.tf [hunk 3]
```diff
@@ -1,38 +1,30 @@
...
 
-data "ilert_schedule" "example" {
-  name = "example"
+resource "ilert_schedule" "example" {
+  name     = "example"
+  timezone = "Europe/Berlin"
+  type     = "STATIC"
+  shift {
+    user  = ilert_user.example.id
+    start = "2023-09-01T08:00"
```
Comment (yacut) [inline examples/escalation_policy/main.tf:13]: same here, use relative date string please

File: examples/schedule/main.tf [hunk 4]
```diff
@@ -12,7 +11,7 @@ resource "ilert_schedule" "example_recurring" {
   type     = "RECURRING"
   schedule_layer {
     name      = "layer1"
-    starts_on = "2022-08-30T00:00"
+    starts_on = "2023-08-30T00:00"
```
Comment (yacut) [inline examples/schedule/main.tf:14]: use `formatdate` and `timestamp ` functions here please

File: examples/schedule/providers.tf [hunk 5]
```diff
@@ -2,12 +2,12 @@ terraform {
   required_providers {
     ilert = {
       source  = "iLert/ilert"
-      version = "~> 1.6"
+      version = "~> 1.7"
```
Comment (yacut) [inline examples/schedule/providers.tf:5]: please bump all required providers version for ilert to `~> 2.0`

File: website/docs/index.html.markdown [hunk 6]
```diff
@@ -29,7 +29,7 @@ resource "ilert_alert_source" "example" {
 
 The following arguments are supported in the `provider` block:
 
-- `api_token` - (Optional) An ilert OAuth / Personal Access Token. When not provided or made available via the `ILERT_API_TOKEN` environment variable, the provider can only access resources available anonymously. Conflicts with `organization`. Make sure to exclude the `Bearer ` prefix.
+- `api_token` - (Optional) An ilert OAuth / Personal Access Token. When not provided or made available via the `ILERT_API_TOKEN` environment variable, the provider can only access resources available anonymously. Conflicts with `organization`. Make sure to exclude the `Bearer` prefix.
```
Comment (yacut) [inline website/docs/index.html.markdown:32]: an empty space after Bearer prefix is necessary

**Review by yacut — CHANGES_REQUESTED** [review]

File: examples/e2e/module/schedule.tf [hunk 7]
```diff
@@ -0,0 +1,11 @@
+resource "ilert_schedule" "this" {
+  name     = var.name
+  timezone = "Europe/Berlin"
+  type     = "STATIC"
+  shift {
+    user  = ilert_user.this.id
+    start = "2024-03-07T15:00:00Z"
```
[the remark this answers is not in the document]
Comment (STLVRTX) [reply examples/e2e/module/schedule.tf:7]: I had this already implemented, but the API doesnt support this.

See https://github.com/iLert/engineering-tasks/issues/574 to understand the problem.

[the remark this answers is not in the document]
Comment (yacut) [reply examples/e2e/module/schedule.tf:7]: than use format function to transform the date string

File: examples/e2e/module/user_phone_number_contact.tf (hunk 2 above)
Comment (STLVRTX) [reply examples/e2e/module/user_phone_number_contact.tf:3]: This is the example phone number for region code `DE` which is also used in the web-ui. It is fetched from `api/regions` if i'm not mistaken.

File: examples/e2e/module/schedule.tf (hunk 7 above)
[the remark this answers is not in the document]
Comment (STLVRTX) [reply examples/e2e/module/schedule.tf:7]: ok, my idea would be to cut off the timezone specification with trimSuffix().
https://developer.hashicorp.com/terraform/language/functions/trimsuffix

[the remark this answers is not in the document]
Comment (STLVRTX) [reply examples/e2e/module/schedule.tf:7]: edit: i could achieve the result with only timeadd, timestamp and formatdate

**Review by yacut — APPROVED** [review]

**Pull request merged**